src/app/clusters/electrical-power-measurement-server/electrical-power-measurement-server.h

```diff
@@ -0,0 +1,113 @@
+    virtual DataModel::List<const Structs::MeasurementAccuracyStruct::Type> GetAccuracy()                        = 0;
+    virtual DataModel::List<const Structs::MeasurementRangeStruct::Type> GetRanges()                             = 0;
+    virtual DataModel::Nullable<int64_t> GetVoltage()                                                            = 0;
+    virtual DataModel::Nullable<int64_t> GetActiveCurrent()                                                      = 0;
+    virtual DataModel::Nullable<int64_t> GetReactiveCurrent()                                                    = 0;
+    virtual DataModel::Nullable<int64_t> GetApparentCurrent()                                                    = 0;
+    virtual DataModel::Nullable<int64_t> GetActivePower()                                                        = 0;
+    virtual DataModel::Nullable<int64_t> GetReactivePower()                                                      = 0;
+    virtual DataModel::Nullable<int64_t> GetApparentPower()                                                      = 0;
+    virtual DataModel::Nullable<int64_t> GetRMSVoltage()                                                         = 0;
+    virtual DataModel::Nullable<int64_t> GetRMSCurrent()                                                         = 0;
+    virtual DataModel::Nullable<int64_t> GetRMSPower()                                                           = 0;
+    virtual DataModel::Nullable<int64_t> GetFrequency()                                                          = 0;
+    virtual DataModel::Nullable<DataModel::List<Structs::HarmonicMeasurementStruct::Type>> GetHarmonicCurrents() = 0;
+    virtual DataModel::Nullable<DataModel::List<Structs::HarmonicMeasurementStruct::Type>> GetHarmonicPhases()   = 0;
+    virtual DataModel::Nullable<int64_t> GetPowerFactor()                                                        = 0;
+    virtual DataModel::Nullable<int64_t> GetNeutralCurrent()                                                     = 0;
+
+protected:
+    EndpointId mEndpointId = 0;
+};
+
+enum class OptionalAttributes : uint32_t
+{
+    kOptionalAttributeRanges          = 0x1,
+    kOptionalAttributeVoltage         = 0x2,
+    kOptionalAttributeActiveCurrent   = 0x4,
+    kOptionalAttributeReactiveCurrent = 0x8,
+    kOptionalAttributeApparentCurrent = 0x10,
+    kOptionalAttributeReactivePower   = 0x20,
+    kOptionalAttributeApparentPower   = 0x40,
+    kOptionalAttributeRMSVoltage      = 0x80,
+    kOptionalAttributeRMSCurrent      = 0x100,
+    kOptionalAttributeRMSPower        = 0x200,
+    kOptionalAttributeFrequency       = 0x400,
+    kOptionalAttributePowerFactor     = 0x800,
+    kOptionalAttributeNeutralCurrent  = 0x1000,
+};
+
+class Instance : public AttributeAccessInterface
+{
+public:
+    Instance(EndpointId aEndpointId, Delegate & aDelegate, Feature aFeature, OptionalAttributes aOptionalAttributes) :
```

Comment:
Should be `BitMask<Feature>`, I would think.  Otherwise you have to pass in values that are not actually valid Feature values.